fdbrpc/include/fdbrpc/LoadBalance.actor.h

```diff
@@ -595,13 +601,15 @@ Future<REPLY_TYPE(Request)> loadBalance(
 				useAlt = (nextAlt + alternatives->size() - 1) % alternatives->size();
 
 			stream = &alternatives->get(useAlt, channel);
+			distance = alternatives->getDistance(useAlt);
 			if (!IFailureMonitor::failureMonitor().getState(stream->getEndpoint()).failed &&
 			    (!firstRequestEndpoint.present() || stream->getEndpoint().token.first() != firstRequestEndpoint.get()))
 				break;
 			nextAlt = (nextAlt + 1) % alternatives->size();
 			if (nextAlt == startAlt)
 				triedAllOptions = TriedAllOptions::True;
 			stream = nullptr;
+			distance = LBDistance::DISTANT;
```

Comment:
can you explain why do we do it here? what happens if code reaches here and we are considering a DISTANT call?

Reply:
This is reset `distance` to a default value. the logic below uses `stream` for processing, so setting a value here has no effect.